packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -3612,7 +3613,9 @@ class _TimePickerDefaultsM3 extends _TimePickerDefaults {
       // TODO(tahatesser): Update this when https://github.com/flutter/flutter/issues/131247 is fixed.
       // This is using the correct text style from Material 3 spec.
       // https://m3.material.io/components/time-pickers/specs#fd0b6939-edab-4058-82e1-93d163945215
-      return _textTheme.displayMedium!.copyWith(color: _hourMinuteTextColor.resolve(states));
+      return entryMode == TimePickerEntryMode.dial
```

Comment:
Maybe make this a switch instead, so that if we add more modes it will flag it.

Reply:
Thank you so much! This actually saved me from making mistake. I totally forgot to handle `dialOnly` and `inputOnly` modes. 

Now it will look like this:
```dart
      return switch (entryMode) {
        TimePickerEntryMode.dial ||  TimePickerEntryMode.dialOnly
          => _textTheme.displayLarge!.copyWith(color: _hourMinuteTextColor.resolve(states)),
        TimePickerEntryMode.input ||  TimePickerEntryMode.inputOnly
          => _textTheme.displayMedium!.copyWith(color: _hourMinuteTextColor.resolve(states)),
      };
```